addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1699,6 +1782,109 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
+                    condition
+                );
+            case "less or equal":
+                return  _t("Invalid: field is not less than or equal to %s.",
+                    condition
+                );
+        }
+
+        let start, end, format;
+
+        if (type === "date") {
+            format = localization.dateFormat;
+            start = formatDate(DateTime.fromSeconds(parseInt(condition), { "format": format }));
+            end =  formatDate(DateTime.fromSeconds(parseInt(between), { "format": format }));
+        } else if (type === "datetime") {
+            format = localization.dateTimeFormat;
+            start = formatDateTime(DateTime.fromSeconds(parseInt(condition), { "format" : format }));
+            end =  formatDateTime(DateTime.fromSeconds(parseInt(between), { "format" : format }));
+        }
+
+        switch(comparator) {
+            case "dateEqual":
+                return _t(
+                    "Entered date or time is not correct! it must be %(value)s(%(format)s).", 
+                    { "start" : start , "format" : format }
+                );
+            case "date!equal":
+                return _t(
+                    "Entered date or time is not correct! it must not be %(value)s(%(format)s).",
+                    { "start" : start, "format" : format }
+                );
+            case "before":
+                return _t(
+                    "Entered date or time is not correct! it must be before %(start)s(%(format)s).",
+                    { "start" : start, "format" : format }
+                );
+            case "after":
+                return _t(
+                    "Entered date or time is not correct! it must be after %(start)s(%(format)s).",
+                    { "start" : start, "format" : format }
+                );
+            case "equal or before":
+                return _t(
+                    "Entered date or time is not correct! it must be before or equal to %(start)s(%(format)s).",
+                    { "start" : start, "format" : format }
+                );
+            case "equal or after":
+                return _t(
+                    "Entered date or time is not correct! it must be after or equal to %(start)s(%(format)s).",
+                    { "start" : start, "format" : format }
+                );
+            case "between":
+                return _t(
+                    "Entered date or time is not correct! it must be within %(start)s and %(end)s(%(format)s).",
+                    { "start" : start, "end" : end, "format" : format }
+                );
+            case "!between":
+                return _t(
+                    "Entered date or time is not correct! it must not be within %(start)s and %(end)s(%(format)s).",
+                    { "start" : start, "end" : end, "format" : format }
+                );
+            default:
+                return  _t("An error has occured, the form has not been sent.");
+        }
+    },
```

Comment:
Maybe something like this. More readable in my opinion.
```suggestion
const messages = {
    "dateEqual": "it must be %(start)s (%(format)s).",
    "date!equal": "it must not be %(start)s (%(format)s).",
    "before": "it must be before %(start)s (%(format)s).",
    "after": "it must be after %(start)s (%(format)s).",
    "equal or before": "it must be before or equal to %(start)s (%(format)s).",
    "equal or after": "it must be after or equal to %(start)s (%(format)s).",
    "between": "it must be within %(start)s and %(end)s (%(format)s).",
    "!between": "it must not be within %(start)s and %(end)s (%(format)s)."
};

if (messages[comparator]) {
    return _t(
        "Entered date or time is not correct! " + messages[comparator],
        { start, end, format }
    );
}

return _t("An error has occurred, the form has not been sent.");
```